### PR TITLE
Add task docs for ansible

### PR DESCRIPTION
**Pull request by captncraig**
Completes #5959

### CODE REVIEW

File: docs/sources/flow/get-started/install/ansible.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+description: Learn how to install Grafana Agent Flow with Ansible
+menuTitle: Ansible
+title: Install Grafana Agent Flow with Ansible
+weight: 550
+---
+
+# Install or uninstall {{% param "PRODUCT_NAME" %}} using Ansible
+
+If your machines are already managed with Ansible, you can install {{< param "PRODUCT_NAME" >}} from
+our package repositories with these steps.
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:14]:
```suggestion
You can use Ansible to install and manage {{< param "PRODUCT_NAME" >}}.
```

We could simplify this and make it more direct.

File: docs/sources/flow/get-started/install/ansible.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+1. Add these tasks to your playbook to add the Grafana package repositories to your system:
+    ```yaml
+    - name: "Install deb repo"
+      when:
+        - "ansible_pkg_mgr == 'apt'"
+      block:
+        - name: "Import Grafana apt gpg key"
+          ansible.builtin.get_url:
+            url: "https://apt.grafana.com/gpg.key"
+            dest: /usr/share/keyrings/grafana.key
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:37]:
```suggestion
            dest: /etc/apt/keyrings/grafana.gpg
```

Suggesting this change to keep it in sync with the install instructions here: https://grafana.com/docs/agent/latest/flow/setup/install/linux/

File: docs/sources/flow/get-started/install/ansible.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+If your machines are already managed with Ansible, you can install {{< param "PRODUCT_NAME" >}} from
+our package repositories with these steps.
+
+## Before you begin
+
+- These steps assume you already have a working [Ansible](https://www.ansible.com/) setup,
+and a pre-existing inventory.
+- You can add the tasks below to any new or existing Role you choose.
+- These tasks install from Grafana's package repositories. They expect to target linux systems using 
+apt or yum package managers.
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:22]:
```suggestion
- These tasks install {{< param "PRODUCT_NAME" >}} from the package repositories. They expect to target Linux systems using 
APT or YUM package managers.
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+## Steps
+
+To add {{% param "PRODUCT_NAME" %}} to a host:
+
+1. Add these tasks to your playbook to add the Grafana package repositories to your system:
+    ```yaml
+    - name: "Install deb repo"
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:30]:
```suggestion
    - name: "Install DEB repo"
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      when:
+        - "ansible_pkg_mgr == 'apt'"
+      block:
+        - name: "Import Grafana apt gpg key"
+          ansible.builtin.get_url:
+            url: "https://apt.grafana.com/gpg.key"
+            dest: /usr/share/keyrings/grafana.key
+            mode: "0644"
+
+        - name: "Add Grafana apt repository"
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:40]:
```suggestion
        - name: "Add Grafana APT repository"
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+To add {{% param "PRODUCT_NAME" %}} to a host:
+
+1. Add these tasks to your playbook to add the Grafana package repositories to your system:
+    ```yaml
+    - name: "Install deb repo"
+      when:
+        - "ansible_pkg_mgr == 'apt'"
+      block:
+        - name: "Import Grafana apt gpg key"
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:34]:
```suggestion
        - name: "Import Grafana APT GPG key"
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+and a pre-existing inventory.
+- You can add the tasks below to any new or existing Role you choose.
+- These tasks install from Grafana's package repositories. They expect to target linux systems using 
+apt or yum package managers.
+
+## Steps
+
+To add {{% param "PRODUCT_NAME" %}} to a host:
+
+1. Add these tasks to your playbook to add the Grafana package repositories to your system:
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:28]:
```suggestion
1. Add the following tasks to your playbook to add the Grafana package repositories to your system:
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+      block:
+        - name: "Import Grafana apt gpg key"
+          ansible.builtin.get_url:
+            url: "https://apt.grafana.com/gpg.key"
+            dest: /usr/share/keyrings/grafana.key
+            mode: "0644"
+
+        - name: "Add Grafana apt repository"
+          ansible.builtin.apt_repository:
+            repo: "deb [signed-by=/usr/share/keyrings/grafana.key] https://apt.grafana.com/ stable main"
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:42]:
```suggestion
            repo: "deb [signed-by=/etc/apt/keyrings/grafana.gpg] https://apt.grafana.com stable main"
```

Again to align with existing Debian/Ubuntu install instructions

File: docs/sources/flow/get-started/install/ansible.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+            state: present
+            update_cache: true
+    - name: "Install yum repo"
+      when:
+        - "ansible_pkg_mgr in ['yum', 'dnf']"
+      block:
+        - name: "Add Grafana yum/dnf repository"
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:49]:
```suggestion
        - name: "Add Grafana YUM/DNF repository"
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+          ansible.builtin.yum_repository:
+            name: grafana
+            description: grafana
+            baseurl: "https://packages.grafana.com/oss/rpm"
+            enabled: true
+            gpgkey: "https://packages.grafana.com/gpg.key"
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:55]:
```suggestion
            baseurl: "https://rpm.grafana.com/oss/rpm"
            enabled: true
            gpgkey: "https://rpm.grafana.com/gpg.key"
```

The info documented here: https://packages.grafana.com/ shows `baseurl=https://rpm.grafana.com`  Does Ansible work with the deeper dir structure or does it need/work with the root URL?

Same path documented for `gpgkey`  Just changing for consistency in all sources... https://packages.grafana.com/ and https://rpm.grafana.com/ and https://grafana.com/docs/agent/latest/flow/setup/install/linux/

File: docs/sources/flow/get-started/install/ansible.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+            repo_gpgcheck: true
+            gpgcheck: true
+    ```
+1. Add these tasks to install and enable the `grafana-agent-flow` service:
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:59]:
```suggestion
1. Add the following tasks to install and enable the `grafana-agent-flow` service:
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+    - name: Enable grafana-agent-flow service
+      ansible.builtin.service:
+        name: grafana-agent-flow
+        enabled: yes
+        state: started
+    ```
+
+## Configuration
+
+The `grafana-agent-flow` package will install a default configuration file that does not send telemetry anywhere.
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:75]:
```suggestion
The `grafana-agent-flow` package installs a default configuration file that doesn't send telemetry anywhere.
```

File: docs/sources/flow/get-started/install/ansible.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+The default config file location is `/etc/grafana-agent-flow.river`. You may replace this file with your own config, or create a new file for the service to use.
```
Comment (clayton-cornell) [inline docs/sources/flow/get-started/install/ansible.md:77]:
```suggestion
The default configuration file location is `/etc/grafana-agent-flow.river`. You can replace this file with your own configuration, or create a new configuration file for the service to use. 
```

**Review by clayton-cornell — COMMENTED** [review]
Some suggestions and questions

**Review by thampiotr — APPROVED** [review]
LGTM with Clayton's suggestions. Thanks!

File: docs/sources/flow/get-started/install/ansible.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+## Configuration
+
+The `grafana-agent-flow` package will install a default configuration file that does not send telemetry anywhere.
+
+The default config file location is `/etc/grafana-agent-flow.river`. You may replace this file with your own config, or create a new file for the service to use. 
+
+## Next steps
+
+- [Configure {{< param "PRODUCT_NAME" >}}][Configure]
```
Comment (thampiotr) [inline docs/sources/flow/get-started/install/ansible.md:81]: I think we should link to https://grafana.com/docs/agent/next/flow/get-started/run/linux/ as "Run".

**Comment by ishanjainn** [comment]
Have you checked grafana-ansible-collection repo if its useful here. We do have built roles for installation of agent already

**Comment by ishanjainn** [comment]
https://grafana.com/docs/grafana-cloud/developer-resources/infrastructure-as-code/ansible/ansible-grafana-agent-linux/

Here's the link to the doc. And configuration might look long but thats just the agent config. If you want it to be small, You just have to remote that section but still you would need the agent config in the host so you would add a seperate task. Both cases work by ansible users generally prefer to keep tasks aligned to a single tasks, Hence the role. Lemme know of you have any questions.

**Comment by ishanjainn** [comment]
> It is built for both static mode and Flow. These docs in this PR must be only for Flow. Because the ansible role supports both, its [configuration ](https://github.com/grafana/grafana-ansible-collection/tree/main/roles/grafana_agent#role-variables)is quite complex. Would be better (and more future-proof when we switch to Flow) to have flow-only role.

Hey, KInda disagree here. Configuration is the list of available variables that users can use but the simplest 101 which you are trying is very small [see](https://grafana.com/docs/grafana-cloud/developer-resources/infrastructure-as-code/ansible/ansible-grafana-agent-linux/)(smaller than the playbook here tbh). And for flow, The configuration is 

```
- hosts: all
  tasks:
    - name: Install Grafana Agent
      ansible.builtin.include_role:
        name: grafana.grafana.grafana_agent
      vars:
        grafana_agent_mode: flow
        # Change config file on the host to .river
        grafana_agent_config_filename: config.river
        # Change config file to be copied
        grafana_agent_provisioned_config_file:  "files/config.river"
        # Remove default flags
        grafana_agent_flags_extra:
          server.http.listen-addr: '0.0.0.0:12345'
```

**Comment by thampiotr** [comment]
Thanks @ishanjainn for sharing this, much appreciated! 

> Hence that might be not possible and agent when does remove static, It would be removed from here aswell so we should have no issues

At some point we will remove the static mode from the latest versions of the Agent Flow, but there still will be users that didn't migrate and are on static mode. Both groups of users may use Ansible, so I wonder if grafana-ansible-collection can remove support for static mode, but still keep the older versions of the collection available for those that didn't migrate? 

Similarly, I wonder what is your breaking changes policy? 

Also, @captncraig, when you're back, I wonder what do you think is the best option forward?

**Comment by ishanjainn** [comment]
> Both groups of users may use Ansible, so I wonder if grafana-ansible-collection can remove support for static mode, but still keep the older versions of the collection available for those that didn't migrate?

Yeah that can be done, Whenever we do that and just keep flow mode in latest versions we can just ket users know if they are using static mode they would have to use a specific version of the collection.

For breaking changes we haven't had a scenario yet where a breaking change in agent breaks the role as the role is just a wrapper around the binary and config (where the chnages are), At max its small fields we add to the role variables but that alaways kept in alignment with agent releases.

**Comment by captncraig** [comment]
If that is what we are recommending in grafana cloud, we should do the same here for consistency. Part of the original idea was to have something simple and standalone, but the snippet in that doc is really nice. If that doc was for flow mode, I would absolutely say this page is redundant, and we should use the same one. 

Perhaps I should make a copy of https://grafana.com/docs/grafana-cloud/developer-resources/infrastructure-as-code/ansible/ansible-grafana-agent-linux/ that uses flow mode? Can include it in our docs for now, and then reference or use it in the grafana-cloud docs as we decide to publish that as the default version.

**Comment by ishanjainn** [comment]
@captncraig, I'll be adding the flow mode instructions to the cloud doc by this week itself(its been long due), Then probably you can use the doc as is. Hope that helps!

**Comment by captncraig** [comment]
Ok, I'm going to close this, with intention of copying your doc in once completed. @ishanjainn can you please update #5959 when that new flow doc is ready?

**Comment by ishanjainn** [comment]
Yup will do! 
Fo reference, The PR to update the ansible doc in cloud docs https://github.com/grafana/website/pull/17894 if you wanna have a look

**Pull request closed without merging**